src/main/java/apoc/atomic/Atomic.java

```diff
@@ -0,0 +1,154 @@
+	public Stream<AtomicResults> subtract(@Name("container") Object container, @Name("propertyName") String property, @Name("number") Number number){
+		PropertyContainer propertyContainer = (PropertyContainer) container;
+		Number oldValue = (Number) propertyContainer.getProperty(property);
+		Number newValue = AtomicUtils.sub(oldValue, number);
+		propertyContainer.setProperty(property, newValue);
+		return Stream.of(new AtomicResults(oldValue,newValue));
+	}
+
+	/**
+	 * concat a property's value
+	 */
+	@Procedure(mode = Mode.WRITE)
+	@Description("apoc.atomic.concat(node/relatonship,propertyName,string) Concats the property's value with the 'string' value")
+	public Stream<AtomicResults> concat(@Name("container") Object container, @Name("propertyName") String property, @Name("string") String string){
+		PropertyContainer propertyContainer = (PropertyContainer) container;
+		String oldValue = propertyContainer.getProperty(property).toString();
+		String newValue = oldValue.concat(string);
+		propertyContainer.setProperty(property,newValue);
+		return Stream.of(new AtomicResults(oldValue,newValue));
+	}
+
+	/**
+	 * insert a value into an array property value
+	 */
+	@Procedure(mode = Mode.WRITE)
+	@Description("apoc.atomic.insert(node/relatonship,propertyName,position,value) insert a value into the property's array value at 'position'")
+	public Stream<AtomicResults> insert(@Name("container") Object container, @Name("propertyName") String property, @Name("position") Long position, @Name("value") Object value) throws ClassNotFoundException {
+		PropertyContainer propertyContainer = (PropertyContainer) container;
+		Object oldValue = propertyContainer.getProperty(property);
+		List<Object> values = insertValueIntoArray(oldValue, position, value);
+		Class clazz = Class.forName(values.toArray()[0].getClass().getName());
+		Object newValue = Array.newInstance(clazz, values.size());
+		try {
+			System.arraycopy(values.toArray(), 0, newValue, 0, values.size());
+		} catch (Exception e) {
+			String message = "Property's array value has type: " + values.toArray()[0].getClass().getName() + ", and your value to insert has type: " + value.getClass().getName();
+			throw new ArrayStoreException(message);
+		}
+		propertyContainer.setProperty(property, newValue);
+		return Stream.of(new AtomicResults(oldValue,newValue));
+	}
+
+	/**
+	 * remove a value into an array property value
+	 */
+	@Procedure(mode = Mode.WRITE)
+	@Description("apoc.atomic.remove(node/relatonship,propertyName,position) remove the element at position 'position'")
+	public Stream<AtomicResults> remove(@Name("container") Object container, @Name("propertyName") String property, @Name("position") Long position)
+			throws ClassNotFoundException {
+		PropertyContainer propertyContainer = (PropertyContainer) container;
+		Object oldValue = propertyContainer.getProperty(property);
+		List<?> values = new ArrayList<>();
+		values.addAll(new ArrayBackedList(oldValue));
```

Comment:
this is way too complicated, just use arraycopy + Arrays.copyOf to shrink, or just a list.